configurate-toml/src/main/java/ninja/leaping/configurate/toml/TOMLConfigurationLoader.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Configurate
+ * Copyright (C) zml and Configurate contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ninja.leaping.configurate.toml;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.moandjiezana.toml.Toml;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.ConfigurationOptions;
+import ninja.leaping.configurate.SimpleConfigurationNode;
+import ninja.leaping.configurate.loader.AbstractConfigurationLoader;
+import ninja.leaping.configurate.loader.CommentHandler;
+import ninja.leaping.configurate.loader.CommentHandlers;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Writer;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A loader for the TOML language, using toml4j.
+ */
+public class TOMLConfigurationLoader extends AbstractConfigurationLoader<ConfigurationNode> {
+
+    @NonNull
+    private final TOMLOptions opts;
+
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends AbstractConfigurationLoader.Builder<Builder> {
+
+        @Nullable
+        private TOMLOptions opts;
+
+        @NonNull
+        public Builder setTomlOptions(@NonNull TOMLOptions opts) {
+            this.opts = opts;
+            return this;
+        }
+
+        @NonNull
+        @Override
+        public TOMLConfigurationLoader build() {
+            return new TOMLConfigurationLoader(this);
+        }
+
+    }
+
+    @NonNull
+    private TOMLConfigurationLoader(@NonNull Builder builder) {
+        super(builder, new CommentHandler[]{CommentHandlers.HASH});
+        if (builder.opts == null) {
+            this.opts = TOMLOptions.builder().build();
+        } else {
+            this.opts = builder.opts;
+        }
+    }
+
+    @Override
+    protected void loadInternal(ConfigurationNode node, BufferedReader reader) throws IOException {
+        Toml toml = new Toml().read(reader);
+        Map<String, Object> map = toml.toMap();
+        fixToml(node, map);
+    }
+
+    @SuppressWarnings("unchecked")
+    void fixToml(@NonNull ConfigurationNode node, @NonNull Map<String, Object> input) {
```

Comment:
`#fixToml` and '#breakToml` aren't very obvious method names, it's not clear what they do.

I'd prefer clearer names, such as read/write, load/save object.